src/pipecat/frames/frames.py

```diff
@@ -558,6 +558,8 @@ class TTSUpdateSettingsFrame(ControlFrame):
     style: Optional[str] = None
     style_degree: Optional[str] = None
     role: Optional[str] = None
+    gender: Optional[str] = None
+    google_style: Optional[str] = None
```

Comment:
I'm wondering if this frame should just be:

```
@dataclass
class TTSUpdateSettingsFrame(ControlFrame):
    settings: Mapping[str, Any]
```

and then use `"gender"`, etc. where needed. Because we might have tons of other settings and by just looking at this frame we have no idea which setting corresponds to each service.

If we do that, maybe we can just have:

```
@dataclass
class ServiceUpdateSettingsFrame(ControlFrame):
    settings: Mapping[str, Any]
```

which is generic for all services. No need for `LLMUpdateSettingsFrame` or `TTSUpdateSettingsFrame`.

I know we lose completions and types but another benefit is that these won't have to change at all and people writing their own services (and not sharing them) they can keep using the frame. Actually, the same idea with `extra`, but we make all `extra`.

Reply:
if we did that, it could be more like the metrics and we could have defined types for each of the various known services. that said, this would be a pretty major breaking change, yeah?

Reply:
i take that back. the metrics aren't really service specific. i mis-remembered. but there could still be some typing there eventually. maybe start with Any for now?

Reply:
i take it back again, we could still have different types defined for llm vs tts vs stt. I'm not sure where the google specific goes, though. Is there a way google.py could expand? or maybe instead of having all settings by [str, Any], there's a `custom_settings` that maps to any? 

```
// in frames.py
@dataclass
class UpdateSettingsFrame(ControlFrame):
    settings: SettingsUpdate

// new settingsUpdates.py ? (name tbd)
class SettingsUpdate(BaseModel):
    model: Optional[str] = None
    custom: Mapping[str, Any]

class LLMSettingsUpdate(SettingsUpdate):
   ...

```

Reply:
btw. working it through with mark, i realized that since these frames are created in client code, the typing described above didn't really make sense.